pkg/loop/internal/relayer/pluginprovider/chainreader/chain_reader.go

```diff
@@ -300,25 +337,54 @@ func (c *Server) QueryKey(ctx context.Context, request *pb.QueryKeyRequest) (*pb
 func (c *Server) Bind(ctx context.Context, bindings *pb.BindRequest) (*emptypb.Empty, error) {
 	tBindings := make([]types.BoundContract, len(bindings.Bindings))
 	for i, b := range bindings.Bindings {
-		tBindings[i] = types.BoundContract{Address: b.Address, Name: b.Name}
+		tBindings[i] = types.BoundContract{
+			Address: b.Address,
+			Name:    b.Name,
+		}
 	}
 
 	return &emptypb.Empty{}, c.impl.Bind(ctx, tBindings)
 }
 
-func getContractEncodedType(contractName, itemType string, possibleTypeProvider any, forEncoding bool) (any, error) {
+func (c *Server) Unbind(ctx context.Context, bindings *pb.UnbindRequest) (*emptypb.Empty, error) {
+	tBindings := make([]types.BoundContract, len(bindings.Bindings))
+	for i, b := range bindings.Bindings {
+		tBindings[i] = types.BoundContract{
+			Address: b.Address,
+			Name:    b.Name,
+		}
+	}
+
+	return &emptypb.Empty{}, c.impl.Unbind(ctx, tBindings)
+}
+
+func getContractEncodedType(readIdentifier string, possibleTypeProvider any, forEncoding bool) (any, error) {
 	if ctp, ok := possibleTypeProvider.(types.ContractTypeProvider); ok {
-		return ctp.CreateContractType(contractName, itemType, forEncoding)
+		val, err := ctp.CreateContractType(readIdentifier, forEncoding)
+
+		return val, err
```

Comment:
nit, why change this
```suggestion
		return ctp.CreateContractType(readIdentifier, forEncoding)
```